app/services/api/v3/actor_node/basic_attributes.rb

```diff
@@ -0,0 +1,214 @@
+          [@actor_quals, @actor_quants, @actor_inds].each do |attribute_hash|
+            attribute_hash.each do |name, attribute|
+              dynamic_attributes[name.downcase] = attribute['value']
+            end
+          end
+          dynamic_attributes
+        end
+
+        def initialize_top_nodes
+          destination_top_nodes = Api::V3::Profiles::TopNodesList.new(
+            @context, @year, @node,
+            other_node_type_name: NodeTypeName::COUNTRY,
+            place_inds: @place_inds,
+            place_quants: @place_quants
+          )
+          @main_destination = destination_top_nodes.sorted_list(
+            @volume_attribute, false, 1
+          ).first
+        end
+
+        def initialize_flow_stats_for_node
+          @flow_stats = Api::V3::Profiles::FlowStatsForNode.new(
+            @context, @year, @node
+          )
+        end
+
+        def exporter_summary
+          # For 1st rank:
+          # Bunge was the largest exporter of soy in BRAZIL in 2015, accounting for 11,061,393 tons.
+          # As an exporter, Bunge sources from 1,136 municipalities, or 44% of the soy production municipalities.
+          # The main destination of the soy exported by Bunge is China, accounting for 50% of the total.
+
+          # For all others:
+          # Cargill was the 2nd largest exporter of soy in BRAZIL in 2015, accounting for 8,801,294 tons.
+          # As an exporter, Cargill sources from 1,224 municipalities, or 48% of the soy production municipalities.
+          # The main destination of the soy exported by Cargill is China, accounting for 66% of the total.
+          text = summary_of_total_trade_volume('exporter')
+          text += summary_of_sources('exporter')
+          text += summary_of_destinations('exporter')
+          text
+        end
+
+        def importer_summary
+          # Bunge was the 1st largest importer of soy from BRAZIL in 2015, accounting for 11,061,393 tons.
+          # As an importer, Bunge sources soy from 1,136 municipalities, or 44% of the soy production municipalities.
+          # The main destination of the soy imported by Bunge is China, accounting for 50% of the total.
+
+          # For all others:
+          # Cargill was the 2nd largest importer of soy from BRAZIL in 2015, accounting for 8,801,294 tons.
+          # As an importer, Cargill sources from 1,224 municipalities, or 48% of the soy production municipalities.
+          # The main destination of the soy imported by Cargill is China, accounting for 66% of the total.
+          text = summary_of_total_trade_volume('importer')
+          text += summary_of_sources('importer')
+          text += summary_of_destinations('importer')
+          text
+        end
+
+        def summary_of_total_trade_volume(profile_type)
+          initialize_trade_volume_for_summary
+          text = "#{@node.name.humanize} was the \
+#{@trade_total_rank_in_country_formatted}largest #{profile_type} of soy \
```

Comment:
there's a space missing, and there's a bug that's causing `@trade_total_rank_in_country_formatted` to not render anything

Reply:
I think the space is not missing here - it's in line 83. It renders nothing when the rank is 1 - that's a feature.

Reply:
I know, but that "feature" is present on all actors :P Just double checked with Amaggi 2015. So it seems my comment should have been on a different line, but the bug is there.

Reply:
yes you are right, shame on me! 🔔 🔔 🔔